### ISSUE TITLE
Error with pstops using manpage examples

### ISSUE THREAD

**Issue by jgclark** [rrthomas/psutils#62]
Me again, sorry.
As you have probably guessed from earlier questions, I'm wanting to use this suite to turn a standard-ly ordered A4 PDF into a version suitable for duplex booklet printing. A combination of `psbook` and `psnup -2` gives me this but requires a printer that can do short-edge duplexing. Mine can't, so I have been trying `pstops`. In the manpage I found what I hoped was the exact answer:

> To re-arrange pages for printing 2-up booklets, use
> "4:-3L@.7(21cm,0)+0L@.7(21cm,14.85cm)" 
> for the front sides, and
> "4:1L@.7(21cm,0)+-2L@.7(21cm,14.85cm)" 
> for the reverse sides (or join them with a comma for duplex printing).

But when I try what I think that means:

`pstops -S "4:-3L@.7(21cm,0)+0L@.7(21cm,14.85cm),4:1L@.7(21cm,0)+-2L@.7(21cm,14.85cm)" in.pdf out.pdf`

I get error
```
pstops: bad page specification:

  PAGESPECS = [MODULO:]SPEC
  SPEC      = [-]PAGENO[@SCALE][L|R|U|H|V][(XOFF,YOFF)][,SPEC|+SPEC]
              MODULO >= 1; 0 <= PAGENO < MODULO
Error: failed to use psnup on in.pdf for ./out.pdf.
```

Running each "half" of the command on their own works.
What am I doing wrong when combining them?

And is the error message accurate? It looks to me like `@SCALE` is in the wrong place from the manpage examples.

**Comment by rrthomas** [comment]
Something is certainly wrong here, as you're doing exactly what the manual says. I'll look into it.

**Comment by rrthomas** [comment]
OK, I think the syntax is correct, and the man page example is missing a crucial detail: looking at the syntax, you can't specify the `MODULO` more than once, at the start, so you need to remove the `4:` from the second string, when you insert the comma. I will fix the example.

**Comment by rrthomas** [comment]
By the way, absolutely no need to apologise. After two complete rewrites I feel I have the code base in half-decent shape, so that fixing bugs feels like a one-way ratchet rather than shifting lumps under a carpet; and of course as when here it's a documentation bug that's doubly easy on me. I shall add a test that uses the exact syntax from the man page.

**Comment by rrthomas** [comment]
I have released v3.3.1 with fixed man page.

**Comment by jgclark** [comment]
Thanks, Reuben, for another quick response.
I should have spotted the double modulo myself.

Incidentally the resulting spec (suitable for long-edge binding) can be simplified to "4:-3L@.7(1w,0h)+0L@.7(1w,0.5h),1L@.7(1w,0h)+-2L@.7(1w,0.5h)". This might be helpful for people who use US letter rather than A4.

I'm now trying to work out how to modify this for short-edge binding. I hoped it might be as simple as "4:-3L@.7(1w,0h)+0L@.7(1w,0.5h),1R@.7(1w,0h)+-2R@.7(1w,0.5h)" (i.e. two of the L get changed for R), but that doesn't give what I want, presumably sending the content off the viewport. The definition of the 'R' code is brief; I'm simply guessing how this interacts with the offsets. Can you clarify, please?

I also tried taking the long-edge output and running pstops on it again to rotate every other page:
`pstops -S "2:0L,1R" test-book-long.pdf test-book-short.pdf`
but that gives a blank set of 4 pages.

**Comment by rrthomas** [comment]
Thanks for the generalization. It won't work for US Letter (the `0.7`s would need to be replaced with other ratios), but will work for any A/B size.

I'll look into clarifying the description of the rest.

**Comment by rrthomas** [comment]
`L` and `R` rotate the page around its bottom left corner. (It is possible this doesn't always happen in PostScript, but I'm not going to worry about that until someone complains.)

Hence, for pages rotated right you need to move them up `0.5h` instead of right `1w`. The incantation is therefore:

`"4:-3L@.7(1w,0h)+0L@.7(1w,0.5h),1R@.7(1w,0h)+-2R@.7(1w,0.5h)"`

I will add this to the docs for now (and clarify how rotation works). Later it will be handled by fixing issue #44.

**Comment by rrthomas** [comment]
Oh also, the error message indeed has the scale in the wrong place, I'm going to make the syntax in the error and that in the man page the same.

**Comment by rrthomas** [comment]
I have released 3.3.2 with the fixes.

**Comment by jgclark** [comment]
Many thanks; I'm confirming that now does just what I was hoping for.
